### PR TITLE
Dark mode: dialog edition

### PR DESCRIPTION
**Pull request by substrae**
image

## ✨ Summary

Contact @substrae with design questions

This PR focuses on `Dialog` and `AlertDialog` dark mode styles. @esizer I'm a tad unsure how to handle dark mode stories in Storybook for this one, because the dialog exists outside of a `<div>` we can wrap. Would love your thoughts

```[tasklist]
### ✅ Requirements
- [ ] `Dialog` has dark mode styles and a dark mode story
- [ ] `AlertDialog` has dark mode styles and a dark mode story
```

### CODE REVIEW

**Comment by esizer** [comment]
I don;t think this PR introduced this issue but it is just more noticeable in dark mode. We have rounding on child elements with a shadow on the parent which is causing a noticeable hard line on the corners.

image 1

**Comment by substrae** [comment]
@esizer Ooof good catch - I didn't notice this. I'll fix it in a second.

**Review by esizer — APPROVED** [review]
LGTM! 👍 

Just had a question about the overlay colour for dark mode.

File: packages/ui/src/components/AlertDialog/AlertDialog.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ const StyledOverlay = React.forwardRef<
 >((props, forwardedRef) => (
   <AlertDialogPrimitive.Overlay
     data-h2-position="base(fixed)"
-    data-h2-background-color="base(black.light.9)"
+    data-h2-background-color="base(black.light.9) base:dark(black.light.9)"
```
[the remark this answers is not in the document]
Comment (substrae) [reply packages/ui/src/components/AlertDialog/AlertDialog.tsx:13]: @esizer it actually ensures it stays the same. Without that bit, `black.light.9` automagically swaps to `white.light.9` due to our dark mode configuration.

**Pull request merged**